File: src/main/resources/static/mitech/assets/js/business/Edificio.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
function Edificio(nome, tassoAggiornamentoColpi, tassoAggiornamentoResistenza, resistenzaLivelloIniziale, colpiLivelloIniziale, tipologia, livelloGiocatore, percentualeDistruzionePunteggio) {

   // this.posizione = posizione;
    this.percentualeDistruzionePunteggio = percentualeDistruzionePunteggio;
    ProprietaOggettoSuCaselle.call(this, nome, tassoAggiornamentoResistenza, resistenzaLivelloIniziale, tipologia,livelloGiocatore,tassoAggiornamentoColpi,  colpiLivelloIniziale);

   
}
Edificio.prototype = Object.create(ProprietaOggettoSuCaselle.prototype);
Object.defineProperty(Edificio.prototype, 'constructor', { 
    value: Edificio, 
    enumerable: true, // so that it does not appear in 'for in' loop
    writable: true });

File: src/main/java/it/univaq/game/domain/Edificio.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package it.univaq.game.domain;
import javax.persistence.Entity;
import javax.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
/**
 *
 * @author Valentina
 */
@Data
@EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
@Entity
@Table(name = "edifici")
public class Edificio extends AbstractEntity<Long> {
    @EqualsAndHashCode.Include
    private String nome;
    private String immagine;
    private String tipologia;
    private int livelloDisponibilita;
    private int tassoAggiornamentoResistenza;
    private int tassoAggiornamentoColpi;
    private int resistenzaLivelloIniziale;
    private int colpiLivelloIniziale;
    private int percentualeDistruzionePunteggio;
}
